### ISSUE TITLE
Random Ultrasonic Spikes

### ISSUE THREAD

**Issue by milespearlman** [Open-STEM/XRP_MicroPython#18]
When the robot was about 30cm away from an object and the ultrasonic was overall reading accurately, the sensor will randomly spike for about 3 cycles, usually + or - about 20cm but sometimes even more. 

![image](https://github.com/Open-STEM/XRP_MicroPython/assets/108246608/72a129e1-b85a-4100-8a0a-706c63f9af81)

![image](https://github.com/Open-STEM/XRP_MicroPython/assets/108246608/ee0ffcc7-0f43-4cac-b7b2-9b998c444817)

**Comment by milespearlman** [comment]
Note that this occurred only when the motors were on.

**Comment by Kreeevin** [comment]
Wait this only happens when the motors are on? can you try to reproduce this bug now that I've fixed some of the issues associated with the encoders?

**Comment by Kreeevin** [comment]
Ok so I've reverted an older change to the rangefinder code to remove the source of this error. The trade off is that the rangefinder code is now back to being blocking, which means that it interrupts the flow of the program for potentially up to 30ms (which should basically never be that extreme). This isn't a permanent solution but it does make the code functional until the other implementation has been fixed.
